sycl-fusion/jit-compiler/lib/translation/KernelTranslation.h

```diff
@@ -0,0 +1,46 @@
+//==- KernelTranslation - Translate SYCL kernels between different formats -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "JITContext.h"
+#include "Kernel.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include <llvm/Support/Error.h>
+#include <vector>
+
+namespace jit_compiler {
+namespace translation {
+
+class KernelTranslator {
+
```

Comment:
Why do we need a class if all of the functions are `static`? Are we expecting any function here depending on some state in the future? I propose keeping the static public functions as functions and moving private members to the `.cpp` as `static` functions.